examples/terraform/hetzner/main.tf

```diff
@@ -54,61 +54,37 @@ resource "hcloud_server" "control_plane" {
   }
 }
 
-resource "hcloud_server_network" "lb" {
-  server_id  = hcloud_server.lb.id
-  network_id = hcloud_network.net.id
+resource "hcloud_load_balancer_network" "load_balancer" {
+  load_balancer_id  = hcloud_load_balancer.load_balancer.id
+  subnet_id = hcloud_network_subnet.kubeone.id
 }
 
-resource "hcloud_server" "lb" {
-  name        = "${var.cluster_name}-lb"
-  server_type = var.lb_type
-  image       = var.image
-  location    = var.datacenter
-
-  ssh_keys = [
-    hcloud_ssh_key.kubeone.id,
-  ]
+resource "hcloud_load_balancer" "load_balancer" {
+  name       = "${var.cluster_name}-lb"
+  load_balancer_type = var.lb_type
+  location   = var.datacenter
 
   labels = {
     "kubeone_cluster_name" = var.cluster_name
     "role"                 = "lb"
   }
-
-  connection {
-    type = "ssh"
-    host = self.ipv4_address
-  }
-
-  provisioner "remote-exec" {
-    script = "gobetween.sh"
-  }
 }
 
-locals {
-  rendered_lb_config = templatefile("./etc_gobetween.tpl", {
-    lb_targets = hcloud_server_network.control_plane.*.ip,
-  })
+resource "hcloud_load_balancer_target" "load_balancer_target" {
+  type             = "server"
```

Comment:
I think that we should use the `label_selector` LB type. We can use these labels, but we can also add new ones if needed

https://github.com/kubermatic/kubeone/blob/dc1cb7ff6378bfda16d04f90f4cb202286514792/examples/terraform/hetzner/main.tf#L51-L54

Reply:
I tried the following:

```
type                   = "label_selector"
label_selector   = "kubeone_cluster_name=${var.cluster_name},role=api"
```

But this will prevent manual adding of servers to the lb targets. Hetzner will automatically add new servers with matching label. For an initial infrastructure state both will do the same.

Reply:
> Hetzner will automatically add new servers with matching label. For an initial infrastructure state both will do the same.

That sounds okay to me. That's the same as we do for other providers as well. I think it's better because if any change happens, it will automatically update the load balancer as well. @kron4eg any opinions?

Reply:
Changes doesn't happen outside of the terraform, supposedly 😅. So it's OK to have `type = "server"`